src/app/shared/filter/filter.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup, FormArray } from '@angular/forms';
+import { MenuController } from '@ionic/angular';
+import urlConfig from 'src/app/config/url.config.json';
+import { ProjectsApiService } from 'src/app/services/projects-api/projects-api.service';
+import { ToastService } from '../../services/toast/toast.service';
+
+@Component({
+  selector: 'app-filter',
+  templateUrl: './filter.component.html',
+  styleUrls: ['./filter.component.scss'],
+})
+export class FilterComponent  implements OnInit {
+  filterList:any = []
+  filterForm: FormGroup = this.fb.group({})
+  @Output() onClose = new EventEmitter();
+
+  constructor(private fb: FormBuilder, private menuControl: MenuController, private apiService: ProjectsApiService,
+    private toastService: ToastService
+  ) { }
+
+  ngOnInit() {
+    this.getFilterList()
+  }
+
+  getFilterList(){
+    this.apiService.get(`${urlConfig.project.filterUrl}?language=en`).subscribe({
+      next: (response: any)=>{
+        if(response.status == 200){
+          this.filterList = response.result
+          this.createForm()
+        }
+      },
+      error: (error:any)=>{
+        this.toastService.presentToast(error.error.message, 'danger')
```

Comment:
Please use chain operator for the error object just in case if the error occurs outside this structure then we should be able to handle that as well so maybe use one or operation as well